metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/TextFunctions.java

```
@@ -0,0 +1,63 @@
+ * (the "License"); you may not use this file except in compliance with the License.  You may obtain
+ * a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.metron.stellar.dsl.functions;
+
+import java.util.List;
+import java.util.Locale;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.text.similarity.FuzzyScore;
+import org.apache.metron.stellar.dsl.BaseStellarFunction;
+import org.apache.metron.stellar.dsl.Stellar;
+
+public class TextFunctions {
+
+  @Stellar(name = "FUZZY_SCORE",
+      description =
+          "Returns the Fuzzy Score which indicates the similarity score between two Strings "
+              +
+              "One point is given for every matched character. Subsequent matches yield two bonus "
+              +
+              "points. A higher score indicates a higher similarity",
+      params = {
+          "string - The full term that should be matched against",
+          "string - The query that will be matched against a term",
+          "string - The IETF BCP 47 language code to use"
+      },
+      returns = "integer representing the score")
+  /**
+   * FuzzyScoreFunction exposes the Apache Commons Text Similarity FuzzyScore through
+   * Stellar.
+   */
+  public static class FuzzyScoreFunction extends BaseStellarFunction {
+
+    @Override
+    public Object apply(List<Object> list) {
+      if (list.size() < 3) {
+        throw new IllegalStateException("FUZZY_SCORE expects three args: [string, string, string]");
+      }
+      String term = (String) list.get(0);
```

If there's an exception here (specifically a classcastexception), we're going to exception.  Given the noise in the data that we have, I would expect this to happen.

I have a couple of questions:
* Do we want to return `NaN` or `Infinity` in that case and log at a warn level (or error level)?
* Do we want to throw an exception which could not possibly be caught in the stream?

I, personally, vote for the first, but I'd like to hear other people's impressions.

---

@nickwallen Check out my latest.  Note that because of the way validate works, we need to return 0 for non-string arguments.

---

Thanks @ottobackwards. That change looks good.